File: programs/reputation/src/instructions/delete_source_account.rs
use anchor_lang::prelude::*;
use crate::state::{source::Source,reputation::Reputation};
pub fn delete_source_account(ctx: Context<DeleteSource>, source_name: String ) -> Result<()> {
    // ctx.accounts.source.create(source_name, ctx.accounts.authority.key(), ctx.accounts.reputation.key())?;
    ctx.accounts.reputation.remove_source()?; 
    // // ctx.accounts.source.bump = ctx.bumps.
    Ok(())
}
#[derive(Accounts)]
#[instruction(source_name: String )]
pub struct DeleteSource<'info> {
    #[account(mut)]
    payer: Signer<'info> ,
    authority: SystemAccount<'info>,
    #[account(mut, seeds=[b"reputation", authority.key().as_ref()], bump,)]
    reputation: Account<'info , Reputation> ,    
    #[account(mut ,close=payer, seeds=[b"source", source_name.as_bytes(), authority.key().as_ref()], bump,)]
    source: Account<'info , Source> ,    
    system_program: Program<'info, System>
}

File: frontend/reputation/app/instructions/deleteSourceAccount.ts
import * as anchor from "@coral-xyz/anchor";
import getRepProgram from "../utils/getRepProgram"

export const deleteSourceAccount = async (
  reputation: anchor.web3.PublicKey,
  authority: anchor.web3.PublicKey,
  payer: anchor.web3.PublicKey,
  sourceName: string
  w: Wallet
) => {
  // Create source account
  const [source] = anchor.web3.PublicKey.findProgramAddressSync(
    [Buffer.from('source'), Buffer.from(sourceName),authority.toBuffer()],
    program.programId
  );
  const deleteSourceTx = await program.methods
    .deleteSourceAccount(
      sourceName,
    )
    .accounts({
      authority,
      reputation,
      source,
      payer,
    })
    .rpc();
  return {deleteSourceTx} 
}

